src/rust/engine/fs/src/lib.rs

```diff
@@ -310,6 +310,20 @@ impl GitignoreStyleExcludes {
       ::ignore::Match::Ignore(_) => true,
     }
   }
+
+  pub fn has_ignored_paths(&self, path: &Path) -> bool {
+    match path.to_str() {
+      None => return false,
```

Comment:
This case would be for non-UTF8 path data... haven't seen it before, and am not sure how likely it actually is. But you might consider renaming this method to indicate uncertainty (i.e. `maybe_parent_of_ignored_path`), and then have this case return `true`, because we can't be sure.